src/redprl/refiner.fun

```diff
@@ -1015,8 +1013,190 @@ struct
       end
   end
 
+  structure Restriction =
+  struct
+    (* This structure provides functions that automate the restriction
+       judgement rules given in "Dependent Cubical Realizability",
+       page 46.
+
+       Such automation is possible because the rules are "syntax
+       directed" on the restriction being performed.
+     *)
+
+    fun neqDims (P.APP P.DIM0, P.APP DIM1) = true
+      | neqDims (P.APP P.DIM1, P.APP DIM0) = true
+      | neqDims _ = false
```

Comment:
Ah, ok! Thanks!

Maybe it could be renamed to something like `dimsContradictory` or something else?